src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -172,11 +172,22 @@ public string GetNodeLocation()
         {
             bool useNode10 = AgentKnobs.UseNode10.GetValue(ExecutionContext).AsBoolean();
             bool taskHasNode10Data = Data is Node10HandlerData;
+            bool taskHasNode16Data = Data is Node16HandlerData;
 
             string nodeFolder = "node";
-            if (taskHasNode10Data || useNode10)
+            if (taskHasNode16Data)
             {
-                Trace.Info($"Task.json has node10 handler data: {taskHasNode10Data}, use node10 for node tasks: {useNode10}");
+                Trace.Info($"Task.json has node16 handler data: {taskHasNode16Data}");
+                nodeFolder = "node16";
+            }
+            else if (taskHasNode10Data)
+            {
+                Trace.Info($"Task.json has node10 handler data: {taskHasNode10Data}");
+                nodeFolder = "node10";
+            }
+            else if (useNode10)
+            {
+                Trace.Info($"Found UseNode10 knob, use node10 for node tasks: {useNode10}");
                 nodeFolder = "node10";
             }
```

Comment:
Nit: Could it make sense to move this logic to a separate private method and call it something like `ResolveNodeVersion`?

Reply:
I tried moving it to a separate function, but only the `Path.combine` logic remains in the original function in this case